### PR TITLE
fix: [CDS-72845]: fixed mapping of store types to ecs overrides

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

https://github.com/harness/harness-core-ui/assets/57372986/4af85dfe-5a7e-43df-b351-e87bc2bdb98b

### CODE REVIEW

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx [hunk 1]
```diff
@@ -302,7 +305,11 @@ export default function useManifestOverride({
           OverrideManifests.OpenshiftParam,
           OverrideManifests.KustomizePatches,
           OverrideManifests.TasVars,
-          OverrideManifests.TasAutoScaler
+          OverrideManifests.TasAutoScaler,
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx:308]: This condition can be moved to utils and used at both places.

File: src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx [hunk 2]
```diff
@@ -316,7 +323,11 @@ export default function useManifestOverride({
           OverrideManifests.OpenshiftParam,
           OverrideManifests.TasManifest,
           OverrideManifests.TasVars,
-          OverrideManifests.TasAutoScaler
+          OverrideManifests.TasAutoScaler,
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx:326]: Same as above

File: src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx [hunk 3]
```diff
@@ -361,6 +372,14 @@ export default function useManifestOverride({
             />
           )
           break
+        case isECSTypeManifest(selectedManifest as ManifestTypes) && manifestStore === ManifestStoreMap.S3:
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx:375]: The other 2 conditions can be written in a similar way.

File: src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx (hunk 1 above)
Comment (sahithikolichala) [reply src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx:308]: ya we can refactor

**Review by tan-nhu — APPROVED** [review]

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverride.tsx [hunk 4]
```diff
@@ -320,7 +323,11 @@ function ServiceManifestOverride({
           OverrideManifests.OpenshiftParam,
           OverrideManifests.KustomizePatches,
           OverrideManifests.TasVars,
-          OverrideManifests.TasAutoScaler
+          OverrideManifests.TasAutoScaler,
+          OverrideManifests.EcsTaskDefinition,
+          OverrideManifests.EcsScalableTargetDefinition,
+          OverrideManifests.EcsScalingPolicyDefinition,
+          OverrideManifests.EcsTaskDefinition
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverride.tsx:330]: `OverrideManifests.EcsTaskDefinition` value is present twice here. Does second value should be `OverrideManifests.EcsServiceDefinition`

**Pull request merged**